server/mysterium_api.go

```diff
@@ -0,0 +1,122 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	log "github.com/cihub/seelog"
+	"github.com/mysterium/node/identity"
+	"github.com/mysterium/node/server/dto"
+	dto_discovery "github.com/mysterium/node/service_discovery/dto"
+	"net/url"
+)
+
+const (
+	mysteriumApiLogPrefix = "[Mysterium.api] "
+)
+
+var mysteriumApiUrl string
+
+type mysteriumApiClient struct {
+	http HttpClient
+}
+
+func NewClient() Client {
+	return &mysteriumApiClient{
+		http: NewJsonClient(
+			mysteriumApiUrl,
+			&http.Client{
+				Transport: &http.Transport{},
+			}),
+	}
+}
+
+func (client *mysteriumApiClient) RegisterIdentity(identity identity.Identity) (err error) {
+	response, err := client.http.DoPost("identities", dto.CreateIdentityRequest{
+		Identity: identity.Address,
+	})
+
+	if err == nil {
+		defer response.Body.Close()
+		log.Info(mysteriumApiLogPrefix, "Identity registered: ", identity)
+	}
+
+	return
+}
+
+func (client *mysteriumApiClient) NodeRegister(proposal dto_discovery.ServiceProposal) (err error) {
+	response, err := client.http.DoPost("node_register", dto.NodeRegisterRequest{
+		ServiceProposal: proposal,
+	})
+
+	if err == nil {
+		defer response.Body.Close()
+		log.Info(mysteriumApiLogPrefix, "Node registered: ", proposal.ProviderId)
+	}
+
+	return
+}
+
+func (client *mysteriumApiClient) NodeSendStats(nodeKey string) (err error) {
+	response, err := client.http.DoPost("node_send_stats", dto.NodeStatsRequest{
+		NodeKey: nodeKey,
+		// TODO Refactor Node statistics with new `SessionStats` DTO
+		Sessions: []dto.SessionStats{},
+	})
+	if err == nil {
+		defer response.Body.Close()
+		log.Info(mysteriumApiLogPrefix, "Node stats sent: ", nodeKey)
+	}
+
+	return nil
```

Comment:
Shouldn't this be `return`, instead `return nil`?
It seems that we are always returning no error, even when error happens.

Reply:
Yeah, I dont se case then error is returned

Reply:
Same applies for some of other endpoints.
This isn't related to this PR, so adding a TODO or Jira ticket seems enough.

Reply:
Will fix this just as example for other endpoints.